### PR TITLE
feat(hackathon-3000): add map component

### PR DESCRIPTION
**Pull request by thmsobrmlr**
## Problem

We want to show a map for the persons feed.

Screenshot 2023-10-24 at 15 36 19

## Changes

This PR:
- adds a feed generic map component based on maplibre-gl (the open source fork of the maptiler v1 js library)
- allows configuring this map with a maplibre style url, which defines the source of map tiles from the environment
- adds this map to the persons feed as a notebook node

It doesn't:
- specify which provider we use for map tiles, so we're not dependent on a decision there
- generalize the map notebook node for usages other than a person, since i wasn't sure how we're resolving dependencies between nodes i.e. how i would set coordinates for a generic map node from the feed node

## How did you test this code?

I've tested the storybook and feed node with a valid maplibre config locally

### CODE REVIEW

File: frontend/src/scenes/notebooks/Nodes/NotebookNodeMap/NotebookNodeMap.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        return null
+    }
+
+    const longtitude = person?.properties?.['$geoip_longitude']
+    const latitude = person?.properties?.['$geoip_latitude']
+    const personCoordinates: [number, number] | null =
+        !isNaN(longtitude) && !isNaN(latitude) ? [longtitude, latitude] : null
+
+    if (!personCoordinates) {
+        return <NotebookNodeEmptyState message="No map available." />
```
Comment (benjackwhite) [inline frontend/src/scenes/notebooks/Nodes/NotebookNodeMap/NotebookNodeMap.tsx:37]: This somewhat conflicts with the <NotFound> component that attempts to achieve a similar thing.

I think in this case it makes sense that is shows something different, but not sure it should be a generic component as it is quite specific to this sort of node.

File: frontend/src/scenes/notebooks/Nodes/NotebookNodeMap/NotebookNodeMap.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+type NotebookNodeMapAttributes = {
+    id: string
+}
+
+export const NotebookNodeMap = createPostHogWidgetNode<NotebookNodeMapAttributes>({
+    nodeType: NotebookNodeType.Map,
+    titlePlaceholder: 'Location',
+    Component,
+    resizeable: false,
```
Comment (benjackwhite) [inline frontend/src/scenes/notebooks/Nodes/NotebookNodeMap/NotebookNodeMap.tsx:57]: Any reason why we wouldn't want this to be resizeable with a default height?

**Review by benjackwhite — APPROVED** [review]
Generally speaking looks great with only some minor comments

File: frontend/src/lib/components/Map/Map.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+type Story = StoryObj<typeof Map>
+
+const coordinates: [number, number] = [0.119167, 52.205276]
+
+export const Unavailable: Story = {}
+
+export const Basic: Story = {
+    render: (args) => (
+        <MapComponent
+            mapLibreStyleUrl="" // TODO: set this value for the publish storybook and visual regression tests
```
Comment (Twixes) [inline frontend/src/lib/components/Map/Map.stories.tsx:20]: TODO left, not sure for when

File: frontend/src/lib/components/Map/Map.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+/** Latitude and longtitude in degrees (+lat is east, -lat is west, +lon is south, -lon is north). */
+export interface MapProps {
+    /** Coordinates to center the map on by default. */
+    center: [number, number]
+    /** Markers to show. */
+    markers?: Marker[]
+    /** Map container class names. */
+    className?: string
+    /** The map's MapLibre style. This must be a JSON object conforming to the schema described in the MapLibre Style Specification, or a URL to such JSON. */
+    mapLibreStyleUrl: string
```
Comment (Twixes) [inline frontend/src/lib/components/Map/Map.tsx:16]: If the type is `string`, then I suppose this cannot be a JSON object as the docstring says, and only a URL?

File: frontend/src/lib/components/Map/Map.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    component: Map,
+    tags: ['autodocs'],
+}
+type Story = StoryObj<typeof Map>
+
+const coordinates: [number, number] = [0.119167, 52.205276]
+
+export const Unavailable: Story = {}
+
+export const Basic: Story = {
```
Comment (Twixes) [inline frontend/src/lib/components/Map/Map.stories.tsx:17]: This story doesn't seem to have a snapshot, while the Unavailable one does

**Comment by benjackwhite** [comment]
Closing in favour of https://github.com/PostHog/posthog/pull/18184 (I'll address the comments over there)

**Pull request closed without merging**